src/python/pants/backend/project_info/dependencies.py

```diff
@@ -45,6 +45,9 @@ def register_options(cls, register):
             "--type",
             type=DependencyType,
             default=DependencyType.SOURCE,
+            removal_version="2.9.0.dev0",
+            removal_hint="This option is misleading and not very useful. In the future there "
+            "will be a more robust way of querying and filtering dependencies.",
```

Comment:
I think the main benefit currently is getting raw requirement strings. Probably we should give the command here to get the equivalent.

Reply:
Which is roughly:
```
./pants filter --target-type=python_requirement_library :: | xargs ./pants peek | jq '.[]["requirements"][]'
```

Reply:
I notice that piping pants commands causes `Initialization options changed: reinitializing scheduler...` every time?

Reply:
I'd rather not burn a tree on this, esp since it's a temporary deprecation notice. But we should have a recipes page in the docs for this kind of thing.

Reply:
Yea. Should probably fix that, but right now whether we use `--colors` and `--dynamic-ui` are determined by whether the output is a TTY, and fingerprinted in the `Scheduler`.

Reply:
Ah well, tree burning...

Reply:
Renders as:

```
13:05:50.26 [WARN] DEPRECATED: option 'type' in scope 'dependencies' will be removed in version 2.9.0.dev0.

This option is misleading and not very useful. In the future there will be a more robust way of querying and filtering dependencies.  Meanwhile you can get the list of requirement strings for a set of targets using something like
./pants dependencies :: \
| xargs ./pants filter --target-type=python_requirement_library \
| xargs ./pants peek | jq -r '.[]["requirements"][]'
```

Reply:
> Yea. Should probably fix that, but right now whether we use `--colors` and `--dynamic-ui` are determined by whether the output is a TTY (vs a pipe), and fingerprinted in the `Scheduler`.

It's kind of a nonstarter to tell users that piping is the blessed way to do this sort of thing, if it's going to kill performance.

Reply:
Thank you for adding this!